src/java/org/apache/cassandra/cql3/conditions/ColumnConditions.java

```diff
@@ -77,6 +79,15 @@ public Collection<ColumnMetadata> getColumns()
                      .collect(Collectors.toList());
     }
 
+    @Override
+    public Set<ColumnMetadata> getAnalyzedColumns(IndexRegistry indexRegistry)
+    {
+        return Stream.concat(columnConditions.stream(), staticConditions.stream())
+                     .filter(c -> indexRegistry.getAnalyzerFor(c.column, c.operator).isPresent())
+                     .map(c -> c.column)
+                     .collect(Collectors.toSet());
```

Comment:
I did it that way for symmetry with the method right above, `getColumns`, which is also using streams in ASF. I've changed both methods to not use streams and made the class `Iterable` to prevent some of the verbosity.